src/main/webapp/app/store/AnnotationStore.ts

```diff
@@ -97,55 +101,7 @@ export class AnnotationStore {
   @observable genomicChangeQuery: string;
   @observable referenceGenomeQuery: REFERENCE_GENOME = REFERENCE_GENOME.GRCh37;
 
-  @computed get cancerTypeFilter() {
-    return this.mutationMapperStore.result
-      ? findCancerTypeFilter(
-          this.mutationMapperStore.result.dataStore.dataFilters
-        )
-      : undefined;
-  }
-
-  @computed get selectedCancerTypes() {
-    if (this.cancerTypeFilter) {
-      return this.cancerTypeFilter.values;
-    } else if (this.selectedPositions.length > 0) {
-      return this.barChartData
-        .filter(
-          data =>
-            data.alterations.filter(alteration =>
-              this.selectedPositions.includes(alteration.proteinStartPosition)
-            ).length > 0
-        )
-        .map(data => data.x);
-    }
-    return [];
-  }
-
-  @computed
-  public get oncogenicityFilter() {
-    return this.mutationMapperStore.result
-      ? findOncogenicityFilter(
-          this.mutationMapperStore.result.dataStore.dataFilters
-        )
-      : undefined;
-  }
-
-  @computed
-  public get oncogenicityFilters() {
-    return this.oncogenicityFilter ? this.oncogenicityFilter.values : [];
-  }
-
-  @computed get positionFilter() {
-    return this.mutationMapperStore.result
-      ? findPositionFilter(
-          this.mutationMapperStore.result.dataStore.selectionFilters
-        )
-      : undefined;
-  }
-
-  @computed get selectedPositions() {
-    return this.positionFilter ? this.positionFilter.values : [];
-  }
```

Comment:
I restructured the content so all computed values are there. I will mention the portion that you need to review.